cdap-examples/WikipediaPipeline/src/main/scala/co/cask/cdap/examples/wikipedia/ClusteringUtils.scala

```diff
@@ -98,15 +98,15 @@ object ClusteringUtils {
    * Store the clustering results in the specified dataset and update accumulators and workflow token.
    */
   def storeResults(sc: SparkContext, sec: SparkExecutionContext,
-                   results: Array[Array[(String, Double)]], tableName: String) = {
+                   results: Array[Array[(String, Double)]], namespace: String, tableName: String) = {
     val numRecords: Accumulator[Int] = sc.accumulator(0, "num.records")
     val initial= new Term("", 0.0)
     val highestScore: Accumulator[Term] = sc.accumulator(initial, "highest.score")(HighestAccumulatorParam)
 
     sec.execute(new TxRunnable {
       override def run(context: DatasetContext) = {
-        val table: Table = context.getDataset(tableName)
-
+        var table: Table = if (namespace != null) context.getDataset(namespace, tableName)
```

Comment:
I think I had responded to the earlier comment about this. I did not find Spark program context getNamespace API. 
Also since this is scala code there is no natural ?: operator used. More at https://www.safaribooksonline.com/library/view/scala-cookbook/9781449340292/ch03s07.html